homeassistant/components/nibe_heatpump/config_flow.py

```diff
@@ -98,9 +121,55 @@ async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str,
     finally:
         await connection.stop()
 
-    return {
-        "title": f"{data[CONF_MODEL]} at {data[CONF_IP_ADDRESS]}",
+    return f"{data[CONF_MODEL]} at {data[CONF_IP_ADDRESS]}", {
+        **data,
         CONF_WORD_SWAP: word_swap,
+        CONF_CONNECTION_TYPE: CONF_CONNECTION_TYPE_NIBEGW,
+    }
```

Comment:
Not that it matters, but the newer `dict` union syntax would probably work ok here